moshi/src/main/java/com/squareup/moshi/JsonUtf8Reader.java

```diff
@@ -464,7 +464,8 @@ private int peekNumber() throws IOException {
     }
 
     // We've read a complete number. Decide if it's a PEEKED_LONG or a PEEKED_NUMBER.
-    if (last == NUMBER_CHAR_DIGIT && fitsInLong && (value != Long.MIN_VALUE || negative)) {
+    if (last == NUMBER_CHAR_DIGIT && fitsInLong && (value != Long.MIN_VALUE || negative) && (value
+        != 0 || !negative)) {
```

Comment:
This is weird wrapping. Our styles do this? I'd prefer it to wrap on the boolean operator.